app/assets/javascripts/course.ts

```diff
@@ -0,0 +1,378 @@
+import { initDragAndDrop } from "./drag_and_drop.js";
+import { fetch, getURLParameter } from "./util.js";
+import { ScrollSpy } from "./scrollspy";
+import { searchQuery } from "./search";
+
+function loadUsers(_baseUrl: string, _status): void {
+    const status = _status || getURLParameter("status");
+    searchQuery.queryParams.updateParam("status", status);
+}
+
+function initCourseMembers(): void {
+    function init(): void {
+        initUserTabs();
+        initLabelsEditModal();
+    }
+
+    function initUserTabs(): void {
+        const userTabs = document.getElementById("user-tabs");
+        if (userTabs !== null) {
+            const baseUrl = userTabs.getAttribute("data-baseurl");
+
+            // Select tab and load users
+            const selectTab = (tab): HTMLElement => {
+                const kebab = document.getElementById("kebab-menu");
+                const status = tab.getAttribute("data-status");
+                const kebabItems = kebab.querySelectorAll<HTMLElement>("li a.action");
+                let anyShown = false;
+                for (const item of Array.from(kebabItems)) {
```

Comment:
```suggestion
                for (const item of kebabItems) {
```
You should be able to loop over a `NodeList` with a `for...of` directly.

Reply:
I thought so too, but the typescript linter thought otherwise.

The solution was to add `DOM.iterable` to the libs in tsconfig. (as suggested in the comment on [this](https://stackoverflow.com/questions/65364900/ts-lint-error-with-document-queryselectorall) stackoverflow post)

Reply:
The problem is that the `target` in our `tsconfig.js` is `es5`. `NodeList` was added in `es6`, so Typescript isn't happy. Since we use Babel I think we could set the target to `es6`, but I'm not completely sure on this.

Reply:
I think we could. Even without babel, es6 should be supported by the browsers we cover.

Reply:
> The problem is that the `target` in our `tsconfig.js` is `es5`. `NodeList` was added in `es6`, so Typescript isn't happy. Since we use Babel I think we could set the target to `es6`, but I'm not completely sure on this.

Why is it a problem that our target is `es5`? I don't immediately find anything online about it and even when we use `es6`as target the linter warning stays without adding the `DOM.iterable` lib

Reply:
I tried https://stackoverflow.com/questions/51723962/typescript-nodelistofelement-is-not-an-array-type-or-a-string-type locally and for me it seemed to work

Reply:
Hmm, it does not work for me (tried both in VSCode and Webstorm to double-check).

Reply:
It also doesn't work for me when setting the target to `es6` and removing `DOM.iterable` from the `lib` list. But I don't see any issue with adding `DOM.iterable` to `lib`?

Reply:
You are right that setting the target itself doesn't work (oops, I modified more locally apparently).

I think the best solution is probably removing `lib` and setting `"target": "es2021"`. Setting the [`target`](https://www.typescriptlang.org/tsconfig#target) will also set a default for [`lib`](https://www.typescriptlang.org/tsconfig#lib), unless you specific it yourself, which is what we did. Since we already include `es2021` libs, we might as well set the target to `es2021` and use the default value.